sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/ExecutorEvaluationStage.java

```diff
@@ -0,0 +1,67 @@
+package com.mesosphere.sdk.offer.evaluate;
+
+import com.mesosphere.sdk.executor.ExecutorUtils;
+import com.mesosphere.sdk.offer.MesosResourcePool;
+import com.mesosphere.sdk.offer.OfferRecommendationSlate;
+import com.mesosphere.sdk.offer.OfferRequirement;
+import org.apache.mesos.Protos;
+
+import java.util.Optional;
+
+/**
+ * This class evaluates an offer against a given {@link OfferRequirement}, ensuring that executor IDs match between
+ * the two and setting the executor ID for a newly-launching pod.
+ */
+public class ExecutorEvaluationStage implements OfferEvaluationStage {
+    private Optional<Protos.ExecutorID> executorId;
```

Comment:
Let's not make fields Optional going forward, but instead have the getters return Optionals.